### PR TITLE
Refactor code

### PR DESCRIPTION
**Pull request by SekoViper**
# Summary
## Implement the following: 

- Refactor code by following the SOLID, DRY, KISS, YAGNI, and ruby style guides.

### CODE REVIEW

File: display_options.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class DisplayOptions
```
Comment (rloterh) [inline display_options.rb:1]: 💠 Nice job factoring display options to be standalone!

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Umair & Seko, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  🏃 🏃‍♂️  

## Highlights 👍  
Good job implementing clean code principles so far!

## Required Changes ♻️
- [x] Nice job, your code is taking shape 👌 However, we can further improve upon our _create_ and _list_ methods found in the `app.rb`, which at present do not adhere to the single responsibility rule. Let's ensure that *all* requests for input and input validation are placed in the `main.rb` file or in classes/methods whose responsibilities are to request or validate input.` The affected methods to consider for further refactoring includes:
  * `create_rental`
  * `ask_student`
  * `ask_teacher`
  * `create_person`
  * `create_book`, and
  * `list_rental`

I urge you to kindly read further on the same via [this link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid).

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

**You can also consider:**
- Kindly include appropriate instructions in your ReadMe on how to run your app, under the _Getting Started_ section.
- Generally, the subheading text _Author1_ and _Author2_ or similar should be replaced with your respective full names in the ReadMe.

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 2]
```diff
@@ -1,92 +1,36 @@
...
-
-    print 'Author: '
-    author = gets.chomp
-
-    @books << Book.new(title, author)
-
-    puts 'Book created successfully'
   end
 
   def create_rental
```
Comment (rloterh) [inline app.rb:37]: - [ ] Your code is gradually taking shape, however, most of the methods are still playing multiple roles, which we want to eliminate in key methods. Endeavor to refactor all requests, for instance `get.chomps`  to be handled in different class/ methods. We can then reference those values which were stored in objects during the request in our actual methods.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Umair & Seko, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦   🏃 🏃‍♂️

## Highlights 👍  
- Lot's of improvements in your code, good job!

## Required Changes ♻️
_Kindly check the inline comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

_Kindly check the inline comments marked **[OPTIONAL]** under the review._

**You can also consider:**
- Kindly include appropriate instructions in your ReadMe on how to run your app, under the _Getting Started_ section.
- Generally, it's good practice that the subheading text _Author1_ and _Author2_ or similar should be replaced with your respective full names in the ReadMe.

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED ✔🍾🎉**
Hello @SekoViper and @umair-rao 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You documented your work professionally.✔
- You asked for the changes in a polite and friendly way.
- Your linters are passing all the checks.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**